src/renderer/components/BackgroundMaps/MapCard.js

```diff
@@ -0,0 +1,104 @@
+// @ts-check
+import * as React from 'react'
+import Button from '@material-ui/core/Button'
+import { makeStyles, Typography } from '@material-ui/core'
+import { useIntl, defineMessages } from 'react-intl'
+import ReactMapboxGl from 'react-mapbox-gl'
+
+import { MAPBOX_ACCESS_TOKEN } from '../../../../config'
+import { convertKbToMb } from '../Settings/BackgroundMaps'
+
+const m = defineMessages({
+  // Abbreviation for megabytes
+  mb: 'MB',
+  // indicates how many offline areas
+  areas: 'offline areas'
+})
+
+/**
+ * @typedef MapCardProps
+ * @prop {import('../Settings/BackgroundMaps').MapServerStyleInfo} offlineMap
+ * @prop {React.Dispatch<React.SetStateAction<import('../Settings/BackgroundMaps').MapServerStyleInfo['id'] | false>>} setMap
+ * @prop {boolean } isBeingViewed
+ */
+
+/** @param {MapCardProps} param */
+export const MapCard = ({ offlineMap, setMap, isBeingViewed }) => {
+  const classes = useStyles()
+  const { formatMessage: t } = useIntl()
+
+  const Mapbox = React.useMemo(
+    () =>
+      ReactMapboxGl({
+        accessToken: MAPBOX_ACCESS_TOKEN,
+        dragRotate: false,
+        pitchWithRotate: false,
+        attributionControl: false,
+        injectCSS: false,
+        interactive: false
+      }),
+    []
+  )
+
+  return (
+    <Button
+      variant='outlined'
+      className={classes.root}
+      onClick={() => setMap(offlineMap.id)}
+    >
+      <div
+        className={classes.inner}
+        style={{ backgroundColor: !isBeingViewed ? '#CCCCD6' : '#0066FF' }}
+      >
+        <div style={{ width: '30%' }}>
+          <Mapbox
+            containerStyle={{
+              height: '100%',
+              width: '100%'
+            }}
+            style={offlineMap.url}
+          />
```

Comment:
Does this look correct? Currently it is not showing any style, just a green box

Reply:
this seems correct! curious about what dataset you're using for your example.

perhaps unrelated, but will note that the way you're instantiating mapbox-react-gl is a little odd. Usually you'd create the instance outside of react i.e. not in a component like you're doing in line 30. See the example from the docs: https://github.com/alex3165/react-mapbox-gl#getting-started

Reply:
Yes creating an instance of Mapbox-gl is expensive (which is what creating an instance of mapbox-react-gl will do) so you should use a single instance for all maps rather than a different instance for each map.

Reply:
So I created one instance for my two components [here](https://github.com/digidem/mapeo-desktop/pull/678/commits/b9623128433155c6dcb90d5f78548a7f711b0a24). Should I extend that further and use that same instance for the `mapview.js` (which is actually instantiated in a  React compnonent and memoized as well, but fixing that is out of scope for this PR)

Reply:
i wouldn't touch that one because

1) out of scope for this PR

2) Wondering if there's some context that i'm missing for that case to instantiate inline. Looks like it relies on a couple of props (basically the access token and the type of screen it's rendered in e.g. map printer), so i'm thinking that that was a pretty intentional (but not ideal) choice. seems like that could be refactored to be improved, but not worth doing in this PR since it's out of scope